controllers/automation/policyPredicate.go

```diff
@@ -21,7 +22,7 @@ var policyPredicateFuncs = predicate.Funcs{
 		// nolint: forcetypeassert
 		plcObjOld := e.ObjectOld.(*policiesv1.Policy)
 
-		return plcObjNew.Status.ComplianceState != plcObjOld.Status.ComplianceState
+		return !cmp.Equal(plcObjNew.Status.Status, plcObjOld.Status.Status)
 	},
```

Comment:
this was main reason of the bug

Reply:
Could you please explain this a bit more? I would think we only want to reconcile when the policy status has changed compliance since it seems that this predicate is only looking at replicated policies.

Reply:
so when 3 clusters noncompliant in a row, at the first cluster, policy's status changes to noncompliant so reconcile only pickup first cluster status.

Reply:
Oh, I see. I misread the code. The predicate is only allowing root policies and not replicated policies. Then you are checking if the compliance per cluster has changed. This makes sense.

Reply:
you right